Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Annotation/AddIshAnnotation.cs

```diff
@@ -0,0 +1,345 @@
+        public string Category { get; set; }
+
+        /// <summary>
+        /// <para type="description">The metadata of the Annotation</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParametersGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectGroup")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "MetadataGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshField[] Metadata { get; set; }
+
+        /// <summary>
+        /// <para type="description">The IShObject - content object containing full LogicalId/Version/Language combination</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = "IshObjectGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshDocumentObj IshObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">The IshAnnotation array that needs to be created. Pipeline</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "IshAnnotationGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshAnnotation[] IshAnnotation { get; set; }
+        
+        protected override void BeginProcessing()
+        {
+            if (IshSession == null) { IshSession = (IshSession)SessionState.PSVariable.GetValue(ISHRemoteSessionStateIshSession); }
+            if (IshSession == null) { throw new ArgumentException(ISHRemoteSessionStateIshSessionException); }
+            WriteDebug($"Using IshSession[{IshSession.Name}] from SessionState.{ISHRemoteSessionStateIshSession}");
+
+            if ((IshSession.ServerIshVersion.MajorVersion < 14) || ((IshSession.ServerIshVersion.MajorVersion == 14) && (IshSession.ServerIshVersion.RevisionVersion < 1)))
+            {
+                throw new PlatformNotSupportedException($"Add-IshAnnotation requires server-side Annotation API which is only available starting from 14.0.1 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+            }
+            base.BeginProcessing();
+        }
+
+        /// <summary>
+        /// Process the Add-IshAnnotation commandlet.
+        /// </summary>
+        /// <exception cref="TrisoftAutomationException"></exception>
+        /// <exception cref="Exception"></exception>
+        /// <remarks>Writes <see cref="IshAnnotation"/> array to the pipeline</remarks>
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                List<IshObject> returnedObjects = new List<IshObject>();
+
+                //1. Add annotations depending on the chosen ParameterSet
+                List<string> returnAnnotations = new List<string>();
+                IshFields returnFields = new IshFields();
+
+                if (ParameterSetName == "MetadataGroup")
```

Comment:
I would recommend use 'switch/case' statement instead of several 'if' statements.